python/pyspark/sql/connect/session.py

```diff
@@ -1060,6 +1068,9 @@ def _start_connect_server(master: str, opts: Dict[str, Any]) -> None:
                 overwrite_conf["spark.connect.grpc.binding.port"] = "0"
 
             origin_remote = os.environ.get("SPARK_REMOTE", None)
+            local_auth_token = str(uuid.uuid4())
+            SparkConnectClient._local_auth_token = local_auth_token
+            os.environ["SPARK_CONNECT_LOCAL_AUTH_TOKEN"] = local_auth_token
```

Comment:
hmm... should it be enabled unconditionally?

currently, there are many spark services have no authN by default, this looks like an exception

Reply:
Actually Py4J gateway server enables it by default :-).